contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    private void deleteDir(FileSystem fs, String target) throws Exception {
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        if (fs.exists(path)) {
+            fs.delete(path, true);
+            if (fs.exists(path)) {
+                warn("Could not delete temp directory " + path + " in CDN!");
+            } else {
+                info("Deleted " + path);
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        String pathFromRoot = target.replaceAll(".*://.*?/", "");
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            FsAction u = perm.getUserAction();
+            FsAction g = perm.getGroupAction();
+            FsAction o = perm.getOtherAction();
+            boolean changed = false;
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYGROUP, false)) {
+                if (!g.implies(FsAction.READ_EXECUTE)) {
+                    g = g.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYOTHER, false)) {
+                if (!o.implies(FsAction.READ_EXECUTE)) {
+                    o = o.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (changed) {
+                FsPermission desiredPerm = new FsPermission(u, g, o);
+                fs.setPermission(parentPath, desiredPerm);
+                assert fs.getFileStatus(parentPath).getPermission().equals(desiredPerm);
+                info("for path " + parent + ", permissions changed:");
+                info("from: " + perm);
+                info("  to: " + desiredPerm);
+            }
+        }
+    }
+
+    private String pickCDN() {
+        List<String> pushClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CLUSTER);
+
+        if (!props.containsKey(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER)) {
+            return "";
+        }
+        List<String> cdnClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER);
+
+        if (pushClusters.size() != cdnClusters.size()) {
+            warn("Cluster sizes are different! Will bypass CDN for push cluster " + destination);
+            return "";
+        }
+
+        int index = pushClusters.indexOf(destination);
+        assert index != -1;
```

Comment:
I read some doc about Java assert:
https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html

Are those assert statements for testing for supposed to execute in runtime as well?
If it is required to run during runtime, it seems we need to explicitly enable it:
By default, assertions are disabled at runtime. Two command-line switches allow you to selectively enable or disable assertions.

Another concern is that AssertionError won't provide any detailed message when it happens, which could make debugging more difficult from my understanding.
What do you think?